### PR TITLE
add flow type to telemetry

### PR DESCRIPTION
**Pull request by Stephen1993**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
1. add `flow_type `filed to telemetry for` pf.flows.test` and `pfazure.flows.create_or_update`.
2. fix get `trace.destination` from `pf._config` when use` PFClient(config={...}) `and `submit a run`.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/_utils/flow_utils.py [hunk 1]
```diff
@@ -335,3 +336,13 @@ def parse_variant(variant: str) -> Tuple[str, str]:
             message=str(error),
             error=error,
         )
+
+
+def get_flow_type(flow_path: Union[str, Path, PathLike]) -> str:
```
Comment (elliotzh) [inline src/promptflow-core/promptflow/_utils/flow_utils.py:341]: this is duplicate to `Run._flow_type`. Is it possible to merge them?

File: src/promptflow-azure/promptflow/azure/operations/_flow_operations.py [hunk 2]
```diff
@@ -603,4 +603,16 @@ def _try_resolve_code_for_flow_to_file_share(cls, flow: Flow, ops: OperationOrch
             flow.path = (Path(path) / flow.path).as_posix()
             flow._code_uploaded = True
 
-    # endregion
```
Comment (elliotzh) [inline src/promptflow-azure/promptflow/azure/operations/_flow_operations.py:606]: removed by mistake?

**Review by elliotzh — APPROVED** [review]

File: src/promptflow-azure/tests/sdk_cli_azure_test/e2etests/test_telemetry.py [hunk 3]
```diff
@@ -466,3 +466,59 @@ def check_evelope():
...
+            assert "flow_type" in custom_dimensions
+            assert custom_dimensions["flow_type"] == flow_type
+
+        with patch.object(PromptFlowSDKExporter, "_log_to_envelope", side_effect=log_event), patch(
+            "promptflow._sdk._telemetry.telemetry.get_telemetry_logger", side_effect=get_telemetry_logger
+        ):
+            flow_type = FlowType.DAG_FLOW
+            try:
+                pf.flows.create_or_update(
+                    flow=FLOWS_DIR / "print_input_flow",
```
Comment (wangchao1230) [inline src/promptflow-azure/tests/sdk_cli_azure_test/e2etests/test_telemetry.py:508]: what if user passed an invalid type obj to flow?

File: src/promptflow-azure/promptflow/azure/operations/_flow_operations.py (hunk 2 above)
Comment (Stephen1993) [reply src/promptflow-azure/promptflow/azure/operations/_flow_operations.py:606]: I have reverted it.

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py [hunk 4]
```diff
@@ -269,13 +269,13 @@ def _validate_column_mapping(cls, column_mapping: dict):
...
     @classmethod
-    def _upload_run_to_cloud(cls, run: Run):
+    def _upload_run_to_cloud(cls, run: Run, config=None):
         error_msg_prefix = f"Failed to upload run {run.name!r} to cloud."
         try:
             from promptflow._sdk._tracing import _get_ws_triad_from_pf_config
             from promptflow.azure._cli._utils import _get_azure_pf_client
 
-            ws_triad = _get_ws_triad_from_pf_config(path=run._get_flow_dir().resolve(), config=run._config)
+            ws_triad = _get_ws_triad_from_pf_config(path=run._get_flow_dir().resolve(), config=config or run._config)
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py:278]: add PR description on this also

File: src/promptflow-azure/tests/sdk_cli_azure_test/e2etests/test_telemetry.py (hunk 3 above)
Comment (Stephen1993) [reply src/promptflow-azure/tests/sdk_cli_azure_test/e2etests/test_telemetry.py:508]: The flow type field will not be added:
image

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py (hunk 4 above)
Comment (Stephen1993) [reply src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py:278]: done

**Review by wangchao1230 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**